### PR TITLE
Fix for random improve stage in coin selection 

### PR DESCRIPTION
**Pull request by lisicky**

### CODE REVIEW

File: rust/src/tx_builder.rs [hunk 1]
```diff
@@ -593,6 +597,7 @@ impl TransactionBuilder {
             .cloned()
             .collect::<Vec<TransactionOutput>>();
         outputs.sort_by_key(|output| by(&output.amount).expect("filtered above"));
+        let mut avaible_coins = by(input_total).unwrap_or(BigNum::zero());
```
Comment (nicolasayotte) [inline rust/src/tx_builder.rs:600]: Typo, should be available_coins

**Comment by nicolasayotte** [comment]
The goal of this PR is to :

1.  avoid the Phase Two of cip2_random_improve_by<F> when doing Strategies 2 and 3 since there is already a separate step with a coin filter that fulfills that function
2. take into account assets already included in previous passes to avoid adding more utxos that intended

File: rust/src/tx_builder.rs (hunk 1 above)
Comment (lisicky) [reply rust/src/tx_builder.rs]: Thanks @nicolasayotte !

**Review by twwu123 — APPROVED** [review]

**Review by lisicky — COMMENTED** [review]
/check

**Review by lisicky — COMMENTED** [review]
/check

**Pull request merged**